tox.ini

```diff
@@ -73,4 +73,10 @@ deps =
     psycopg2-binary
     -r{toxinidir}/requirements.txt
 commands =
+    # Download patroni resource to use in the charm deployment.
+    sh -c 'stat patroni.tar.gz > /dev/null 2>&1 || curl "https://github.com/zalando/patroni/archive/refs/tags/v2.1.3.tar.gz" -L -s > patroni.tar.gz'
```

Comment:
Could this be some sort of fixture, or downloaded in the pytest code using urllib then cleaned up after the test?

The benefit here, is that once you have an uploaded version on chamrhub, you can fetch the resource straight from Charmhub using the API and get a bit more certainty in the test? (Perhaps a future PR?)

Reply:
Yes, it can be, Jon. Nice suggestion! I'll create a new PR to change this that and upload the resource to Charmhub to have this improved flow of testing with the resource. I created an [issue](https://github.com/canonical/postgresql-operator/issues/5) to remind about it.